### PR TITLE
NSM connection (VIPs and routes) update in TAPA

### PR DESCRIPTION
**Pull request by LionelJouin**
The initial NSM request does not contain any information about the ip context. Once the conduit connected (sucessful NSM request), the VIP watcher will call SetVIPs which will update (new request) the NSM connection (SrcIPs and policies).

The initial SrcIPs are saved in the conduit struct (to not mix the target IPs and VIPs).
    
The previous code has been removed.

The last privileges required (NET_ADMIN) has been removed.

issue: #164 
related NSM issues: 
* https://github.com/networkservicemesh/api/issues/119
* https://github.com/networkservicemesh/api/issues/127
* https://github.com/networkservicemesh/sdk/issues/1231
* https://github.com/networkservicemesh/sdk-kernel/issues/426

Changes required before merging:
* https://github.com/Nordix/Meridio/pull/188
* https://github.com/networkservicemesh/api/pull/129
* https://github.com/networkservicemesh/sdk-kernel/pull/425
* https://github.com/networkservicemesh/sdk/pull/1234 (https://github.com/networkservicemesh/sdk/pull/1238)
* https://github.com/networkservicemesh/sdk-kernel/pull/433 (https://github.com/networkservicemesh/sdk-kernel/pull/437)

### CODE REVIEW

**Comment by uablrek** [comment]
What version of NSM is required? Not v1.2.0 I assume?

**Comment by LionelJouin** [comment]
It requires the latest version in the main branch of the nsmgr and forwarder-vpp + this PR: https://github.com/networkservicemesh/sdk-kernel/pull/433. Also, the NSM dependencies of Meridio must be updated (will be done in a separate PR when NSM 1.3 will be available).

**Comment by uablrek** [comment]
Can't say much about the code since my NSM knowledge is limited but I have applied this PR, built and tested and it seem to work fine.

The `forwarder-ovs` was used since `forwarder-vpp` is broken in NSM `v1.3.0-rc.1`.

File: pkg/ambassador/tap/conduit/conduit.go [hunk 1]
```diff
@@ -174,7 +172,7 @@ func (c *Conduit) GetConduit() *ambassadorAPI.Conduit {
 // GetStreams returns the local IPs for this conduit
```
Comment (zolug) [inline pkg/ambassador/tap/conduit/conduit.go:172]: GetIPs

File: pkg/ambassador/tap/conduit/conduit.go [hunk 2]
```diff
@@ -174,7 +172,7 @@ func (c *Conduit) GetConduit() *ambassadorAPI.Conduit {
...
 func (c *Conduit) GetIPs() []string {
 	if c.connection != nil {
-		return c.connection.GetContext().GetIpContext().GetSrcIpAddrs()
```
Comment (zolug) [inline pkg/ambassador/tap/conduit/conduit.go:177]: If it wasn't for conduit_test this function wouldn't be public, as it accesses the connection and thus is expected to be invoked with held mu lock. Currently not a problem, but might be in the future...

File: pkg/ambassador/tap/conduit/conduit.go [hunk 3]
```diff
@@ -186,34 +184,65 @@ func (c *Conduit) SetVIPs(vips []string) error {
...
+		}
+		c.connection.Context.IpContext.Policies = append(c.connection.Context.IpContext.Policies, newPolicyRoute)
 	}
-	// delete remaining vips
-	vipsSlice := []*virtualIP{}
-	for _, vip := range currentVIPs {
-		vipsSlice = append(vipsSlice, vip)
+	var err error
+	// update the NSM connection
+	// TODO: retry if error returned?
```
Comment (zolug) [inline pkg/ambassador/tap/conduit/conduit.go:223]: maybe nsm's retry client could come handy here

**Review by zolug — APPROVED** [review]

**Review by zolug — APPROVED** [review]

**Pull request merged**